src/Widgets/DeviceManagerWidget.vala

```diff
@@ -52,6 +54,29 @@ public class Sound.Widgets.DeviceManagerWidget : Gtk.Grid {
         update_showable ();
     }
 
+    private void disconnected () {
+        if (null == device_list) {
+            return;
+        }
+        foreach (unowned var child in device_list.get_children ()) {
+            device_list.remove (child);
```

Comment:
Might be worth checking whether this is enough to reduce references on the child to zero (at thereby destroy it) otherwise you may need to call its `destroy ()` function. You put debugging code in `~Device ()` to check whether it is getting destroyed.

Reply:
Thanks for the hint! I did a lot of digging and experimenting and got each of the components correctly destroying.